### DISCUSSION TITLE
No documentation about how use the new hooks to add new columns in admin orders

### DISCUSSION THREAD

**Discussion by racmanuel** [woocommerce/woocommerce#43179] [WooCommerce HPOS]
I have the next code: 
```
$this->loader->add_filter('manage_edit-shop_order_columns', $orders, 'my_plugin_woocommerce_add_new_column_shop_order');
$this->loader->add_action('manage_shop_order_posts_custom_column', $orders, 'my_plugin_woocommerce_add_data_to_new_column_shop_order');
```
and this code before active HPOS works but when i active the HPOS orders not work: 
![image](https://github.com/woocommerce/woocommerce/assets/25874893/6d23a093-d1dc-4428-a6ff-44fc9947decb)

I found this answer in the support forum of WooCommerce in WordPress.org: https://wordpress.org/support/topic/add-columns-to-wc-order-page/ 

I solve the problem active the compability mode but i think this feature is very important add to the documentation of the HPOS orders because some plugins show custom columns in orders. 

![image](https://github.com/woocommerce/woocommerce/assets/25874893/a9268369-c760-4b76-9b87-f757219a5eda)

I think is important add this hooks to the HPOS Order docs:
```
woocommerce_shop_order_list_table_columns ($columns)
woocommerce_shop_order_list_table_custom_column ($column, $order)
```

Maybe with some example of how active the old hooks if the HPOS Order is deactive and how active the hooks if the HPOS order is  active.

**Comment by lsinger** [comment]
Hey @racmanuel, thanks for your report. For now we've opened up an issue to cross-reference the hooks in the comments: 

https://github.com/woocommerce/woocommerce/issues/43846

Do you have further suggestions on how and where we should improve this documentation?

**Reply by racmanuel** [reply]
I think it would be best if they were added in this part as a message for the user explaining that if you have custom columns in WooCommerce orders it is necessary to make the adjustment to HPOS and perhaps an example of the use of the Hook and what parameters the function has.

**Reply by lsinger** [reply]
Sounds good, added that to the issue, @racmanuel. 👍

**Comment by ashik4339** [comment]
The classic `manage_edit-shop_order_columns` / `manage_shop_order_posts_custom_column` pair (post-based orders) still work in legacy mode, but with HPOS enabled the orders screen is a different list table class entirely, so it needs different hooks:

```php
// Add a column (order_type is usually 'shop_order')
add_filter( 'woocommerce_shop_order_list_table_columns', function( $columns ) {
    $columns['customer_note'] = __( 'Note', 'my-plugin' );
    return $columns;
} );

// Render its content
add_action( 'woocommerce_shop_order_list_table_custom_column', function( $column, $order ) {
    if ( 'customer_note' === $column ) {
        echo esc_html( $order->get_customer_note() );
    }
}, 10, 2 );
```

The important gotcha: the custom-column action hands you a **`WC_Order` object** as `$order`, not a post ID like the legacy hook did — so `get_post_meta( $order_id, ... )` won't work here, you need `$order->get_meta( '_key' )` or the relevant getter. That trips people up when copy-pasting old snippets into HPOS.

Both hooks are order-type-aware (`woocommerce_{$order_type}_list_table_columns`), so if you're targeting refunds or another custom order type, swap `shop_order` accordingly.
